zowe-rest-api-commons-spring/zossrc/secur.c

```diff
@@ -67,3 +75,38 @@ JNIEXPORT jint JNICALL Java_org_zowe_commons_zos_security_jni_Secur_getLastErrno
 {
     return lastErrno2;
 }
+
+JNIEXPORT jint JNICALL Java_org_zowe_commons_zos_security_jni_Secur_setApplid(JNIEnv *env, jobject obj, jstring jApplid)
```

Comment:
sorry for the questions, but im trying to learn :) 

since there is code to set this, is there no need to ever reset or remove this for a thread?  or does the thread go away after this is set?

Reply:
Good question. It does not need to be unset. Ideally, I would like to set it for all threads of the application but we need to set it before the first authentication call in the thread.